src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PaymentMethodsScreen.tsx

```diff
@@ -235,6 +250,58 @@ const GenericError = () => {
   );
 };
 
+export const useOnRecurringPaymentReceived = ({
+  recurringPaymentId,
+  callback,
+}: {
+  recurringPaymentId: number | undefined;
+  callback: () => void;
+}) => {
+  const {userId} = useAuthContext();
+
+  const mapRecurringPayment = (
+    recurringPayment: FirebaseFirestoreTypes.DocumentData,
+  ): RecurringPayment => {
+    return {
+      id: recurringPayment.id,
+      expires_at: recurringPayment.expires_at,
+      masked_pan: recurringPayment.masked_pan,
+      payment_type: recurringPayment.payment_type,
+    };
+  };
```

Comment:
Do we need to map the recurring payment? Seems a little strange to map the `expires_at`, `masked_pan`, etc., just for looking up the id.